src/ResourceManager/RecoveryServices.Backup/Commands.RecoveryServices.Backup.Providers/Providers/AzureSqlPsBackupProvider.cs

```diff
@@ -188,7 +188,25 @@ public ProtectionPolicyResponse ModifyPolicy()
 
         public List<Models.AzureRmRecoveryServicesBackupContainerBase> ListProtectionContainers()
         {
-            throw new NotImplementedException();
+            string name = (string)this.ProviderData.ProviderParameters[ContainerParams.Name];
+
+            ProtectionContainerListQueryParams queryParams = new ProtectionContainerListQueryParams();
+
+            queryParams.BackupManagementType = Microsoft.Azure.Management.RecoveryServices.Backup.Models.BackupManagementType.AzureSql.ToString();
+
+            var listResponse = HydraAdapter.ListContainers(queryParams);       
+    
+            List<AzureRmRecoveryServicesBackupContainerBase> containerModels = ConversionHelpers.GetContainerModelList(listResponse);
+
+            if (!string.IsNullOrEmpty(name))
+            {
+                if (containerModels != null)
+                {
+                    containerModels = containerModels.Where(x => x.Name == name).ToList();
```

Comment:
Lambda expression are best used with such variables. because it is understood here that x here means containerModels. do let me know if you disagree